specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+                break;
+            case IDNO:
+                m_cached_permissions[cached_key] = false;
+                state = COREWEBVIEW2_PERMISSION_STATE_DENY;
+                break;
+            default:
+                state = COREWEBVIEW2_PERMISSION_STATE_DEFAULT;
+                break;
+            }
+        }
+
+        CHECK_FAILURE(args->put_State(state));
+        CHECK_FAILURE(deferral->Complete());
+    });
+    return S_OK;
+}
+
+```
+
+## Filter Notifications from a specific doamin and send local toast
+
+Learn more about sending a local toast [Send a local toast notification from a C# app - Windows apps | Microsoft Learn](https://learn.microsoft.com/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp).
+
+### C# Sample
+```csharp
+using Microsoft.Toolkit.Uwp.Notifications;
+...
+void WebView_NotificationReceived(object sender, CoreWebView2NotificationReceivedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    var notification = args.Notification;
+
+    // Show notification with local MessageBox
+    string message = "Received notification from " + args.Uri + " with body " + notification.Body;
+    MessageBox.Show(message);
+
+    // Requires Microsoft.Toolkit.Uwp.Notifications NuGet package version 7.0 or greater
+    var toastContent = new ToastContentBuilder()
+        .AddArgument("conversationId", m_id)
+        .AddText("Notification sent from " + args.Uri +":\n")
+        .AddText(notification.Body);
+    toastContent.Show();
+    // See
+    // https://learn.microsoft.com/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp
+    // for how to handle toast activation
+
+    notification.Show();
+    notification.Close();
+    args.Handled = true;
+}
+```
+
+### C++ Sample
+```cpp
+...
+{
+    //! [NotificationReceived]
+    // Register a handler for the NotificationReceived event.
+    CHECK_FAILURE(m_webView2_17->add_NotificationReceived(
+        Callback<ICoreWebView2NotificationReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2NotificationReceivedEventArgs* args) -> HRESULT
+            {
+                // Block notifications from specific URIs and set Handled to
+                // false so the the default notification UI will not be
+                // shown by WebView2 either.
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri);
+                if (ShouldBlockUri(uri.get()))
+                {
+                    CHECK_FAILURE(args->put_Handled(FALSE));
+                }
+                ShowNotification(args);
+                return S_OK;
+            })
+            .Get(),
+        &m_notificationReceivedToken));
+    //! [NotificationReceived]
+    }
+...
+//! [OnNotificationReceived]
+void SettingsComponent::ShowNotification(
+    ICoreWebView2NotificationReceivedEventArgs* args)
```

Comment:
Not saying we have to make a change here necessarily but it strikes me a bit odd that we have a method ShowNotification but it has to take the event args rather than the notification object because we want the URI. Should the URI go on the notification? Or is it just a quirk of this sample and URI shouldn't really be associated with notification?